cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -724,6 +732,17 @@ private V1ObjectMeta createJob(V1ObjectMeta metadata, Map<String, RuntimeSpecifi
       if (e.getCode() == HttpURLConnection.HTTP_CONFLICT) {
         LOG.warn("The kubernetes job already exists : {}. Ignoring resubmission of the job.", e.getResponseBody());
       } else {
+        // Clean up config maps
+        LOG.info("Deleting configmaps for runId {} in namespace {}", twillRunId.getId(), programRuntimeNamespace);
+        try {
+        coreV1Api.deleteCollectionNamespacedConfigMap(programRuntimeNamespace, null, null, null, null, null,
```

Comment:
creation and deletion of the configmap should happen in the same place. If they're split like this, it means methods need to be called in a specific order in order for things to function properly. for example, something more like:

```
createConfigMap()
try {
  createJob();
} catch (...) {
  // delete config map here
}
```